### PR TITLE
[docs] Fix some links to date picker docs

### PR DESCRIPTION
**Pull request by oliviertassinari**
The /components/ folder looked wrong, this seems to be a safe change. 

---

Two thoughts:

1. The data grid entry point has a flat folder structure: https://master--material-ui-x.netlify.app/x/react-data-grid/ but the date picker doesn't: https://master--material-ui-x.netlify.app/x/react-date-pickers/getting-started/. The former seems more efficient from an SEO perspective. What if we added an overview page too for the date picker?
2. Where should the npm `@mui/x-date-picker` package "homepage" link to? `/x/react-date-pickers/` or the specific component. I kind of like the former for suggesting it's a broad package, date picker, time picker, etc.

### CODE REVIEW

File: packages/x-date-pickers/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "bugs": {
     "url": "https://github.com/mui-org/material-ui-x/issues"
   },
-  "homepage": "https://mui.com/components/x/react-date-pickers/date-picker/",
+  "homepage": "https://mui.com/x/react-date-pickers/",
```
Comment (oliviertassinari) [inline packages/x-date-pickers/package.json:11]:
```suggestion
  "homepage": "https://mui.com/x/react-date-pickers/date-picker/",
```

**Review by flaviendelangle — APPROVED** [review]

**Comment by flaviendelangle** [comment]
@oliviertassinari 

I did not create a pickers home page for now. The closest we have is the [Getting Started](https://deploy-preview-4223--material-ui-x.netlify.app/x/react-date-pickers/getting-started/) page.
But we should probably create one in the future yes.

File: packages/x-date-pickers/src/StaticDateTimePicker/StaticDateTimePicker.tsx [hunk 2]
```diff
@@ -38,7 +38,7 @@ type StaticDateTimePickerComponent = (<TDate>(
  *
  * Demos:
  *
- * - [Date Time Picker](https://mui.com/components/x/react-date-pickers/date-time-picker/)
+ * - [Date Time Picker](https://mui.com/x/react-date-pickers/date-time-picker/)
```
Comment (oliviertassinari) [inline packages/x-date-pickers/src/StaticDateTimePicker/StaticDateTimePicker.tsx:41]: In MUI Core, I believe these lines of code are generated automatically and kept in sync in each PR.

Comment (flaviendelangle) [reply packages/x-date-pickers/src/StaticDateTimePicker/StaticDateTimePicker.tsx:41]: We will probably have improvements to do for this part indeed

**Pull request merged**